test/message-generator/messages/common_messages.json

```diff
@@ -48,6 +48,22 @@
             },
             "id": "setup_connection_job_declarator"
         },
+        {
+            "message": {
+                "type": "SetupConnection",
+                "protocol": 1,
+                "min_version": 2,
+                "max_version": 2,
+                "flags": 6,
+                "endpoint_host": "",
+                "endpoint_port": 0,
+                "vendor": "",
+                "hardware_version": "",
+                "firmware": "",
+                "device_id": ""
+            },
+            "id": "setup_connection_job_declarator_with_no_async_flag"
```

Comment:
can you explain the rationale on this?
why is `"flags": 6`?

Reply:
Okay, so the idea is that a setupConnection request from JDC to JDS needs to have a flag with the 0th bit set. This flag value have both the 1st and 2nd bits set but not the 0th. In such a case, the expected behavior is a setupConnection error.

Reply:
why not simply `0`?

1st and 2nd bits will be shifted away and ignored inside `check_flags`, so it feels like there's no real upside for using `6` here, it just makes it more confusing to understand